src/BuildScriptGenerator/DefaultBuildScriptGenerator.cs

```diff
@@ -147,7 +147,7 @@ internal class DefaultBuildScriptGenerator : IBuildScriptGenerator
                 platformVersion = ctx.LanguageVersion;
                 if (string.IsNullOrEmpty(platformVersion))
                 {
-                    var detectionResult = userSuppliedPlatform.Detect(ctx.SourceRepo);
+                    var detectionResult = userSuppliedPlatform.Detect(ctx);
```

Comment:
We need this access to context for example when we want to get information from the build properties.

Reply:
can you give an example of where we'd need the entire context here?

Reply:
Build properties for example, in case of dotnet core platform trying to get information from a build property like `project` which is then used by a component called projectfileprovider. If the provider finds the 'project' property, it will use it.

Reply:
Makes sense. Thanks